arch/arm/core/cortex_m/Kconfig

```diff
@@ -135,7 +135,7 @@ config CPU_CORTEX_M_HAS_VTOR
 config CPU_CORTEX_M_HAS_SPLIM
 	bool
 	# Omit prompt to signify "hidden" option
-	depends on ARMV8_M_MAINLINE
+	depends on ARMV8_M_MAINLINE || (ARMV8_M_SE && !ARM_NONSECURE_FIRMWARE)
```

Comment:
Selecting ARM_NONSECURE_FIRMWARE does not remove SPLIM from the CPU.
IMHO we should introduce a "USE_SPLIM" option, which will be enabled if  (ARMV8_M_SE && !ARM_NONSECURE_FIRMWARE) || ARMV8_M_MAINLINE

Reply:
Interesting comment.

However, I do not agree, actually. Selecting ARM_NONSECURE_FIRMWARE actually _does_ remove SPLIM; note that in BASELINE implementations, SPLIM registers have only secure physical instances; keep in mind that SPLIM registers are fully banked. So an image that is intended to run in non-secure state, cannot use the feature.

Introducing a <...USE_SPLIM> nomenclature kind of brakes the convention used so far, i.e. iuse CPU_HAS_xxx for MCU capabilities.

Reply:
What about introducing CPU_CORTEX_M_HAS_SECURE_SPLIM and CPU_CORTEX_M_HAS_NONSECURE_SPLIM instead of mixing SW and HW worlds?

Reply:
the cores that have splim will also have a mpu, so the dependency will always be met, most likely.  But this makes it clear.

Reply:
Now I see your point